src/main/java/com/apifest/doclet/Doclet.java

```diff
@@ -82,188 +104,333 @@
-
-        defaultFilterClass = System.getProperty("defaultFilterClass");
-
-        outputFile = System.getProperty("mapping.filename");
-
-        applicationPath = System.getProperty("application.path");
-
         System.out.println("Start ApiFest Doclet>>>>>>>>>>>>>>>>>>>");
         System.out.println("mapping.version is: " + System.getProperty("mapping.version"));
-        System.out.println("backend.host: " + System.getProperty("backend.host"));
-        System.out.println("backend.port: " + System.getProperty("backend.port"));
 
         try {
+            List<ParsedEndpoint> parsedEndpoints = new ArrayList<ParsedEndpoint>();
             ClassDoc[] classes = root.classes();
             for (ClassDoc clazz : classes) {
                 MethodDoc[] mDocs = clazz.methods();
                 for (MethodDoc doc : mDocs) {
-                    MappingEndpoint endpoint = getMappingEndpoint(doc);
-                    if(endpoint != null) {
-                        endpoints.add(endpoint);
+                    ParsedEndpoint parsed = parseEndpoint(doc);
+                    if (parsed != null) {
+                        parsedEndpoints.add(parsed);
                     }
                 }
             }
-
-            generateMappingFile(outputFile);
-        } catch (JAXBException e) {
+            if (docletMode.contains(DocletMode.DOC)) {
+                generateDocsFile(parsedEndpoints, mappingDocsOutputFile);
+            }
+            if (docletMode.contains(DocletMode.MAPPING)) {
+                generateMappingFile(parsedEndpoints, mappingOutputFile);
+            }
+            return true;
+        } catch (ParseException e) {
             System.out.println("ERROR: cannot create mapping file, " + e.getMessage());
             return false;
-        } catch (ParseException e) {
+        } catch (JsonGenerationException e) {
+            System.out.println("ERROR: cannot create mapping file, " + e.getMessage());
+            return false;
+        } catch (JsonMappingException e) {
+            System.out.println("ERROR: cannot create mapping file, " + e.getMessage());
+            return false;
+        } catch (IOException e) {
+            System.out.println("ERROR: cannot create mapping file, " + e.getMessage());
+            return false;
+        } catch (JAXBException e) {
             System.out.println("ERROR: cannot create mapping file, " + e.getMessage());
             return false;
         }
-        return true;
     }
 
-    private static void generateMappingFile(String outputFile) throws JAXBException {
-        if(outputFile == null || outputFile.length() == 0 || NULL.equals(outputFile)) {
-            outputFile = String.format(DEFAULT_MAPPING_NAME, mappingVersion);
+    private static void validateConfiguration() {
+        String modeInput = System.getProperty("mode");
+        if (modeInput == null) {
+            throw new IllegalArgumentException("mode is invalid.");
```

Comment:
Here the message may contain a message that mode is mandatory. And that makes me think whether the mode should be mandatory - we can use "mapping" mode as a default one.